ngx-fudis/projects/ngx-fudis/src/lib/components/fudis-checkbox/fudis-checkbox.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, Input, ViewEncapsulation, ViewChild, ElementRef, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+
+@Component({
+	selector: 'fudis-native-checkbox',
+	templateUrl: './fudis-checkbox.component.html',
+	styleUrls: ['./fudis-checkbox.component.scss'],
+	encapsulation: ViewEncapsulation.None,
+})
+export class FudisCheckboxComponent implements OnInit {
+	@ViewChild('fudisCheckboxInput') input: ElementRef<HTMLInputElement>;
+
+	@Input() disabled: boolean = false;
+
+	@Input() disabledInput: string;
+
+	@Input() invalid: string;
+
+	@Input() label: string;
+
+	@Input() name: string;
+
+	@Input() required: boolean;
+
+	@Input() checked?: boolean = false;
+
+	showError: boolean = false;
+
+	control = new FormControl('', Validators.required);
```

Comment:
Tämä on myös periaatteessa turha, koska et tee mitään tarkistuksia tämän formControllin avulla. Kaikki tarkistukset käyttää _this.required_, _this.disabled_ ja _this.checked_ arvoja.

Eli jos tämän poistaa, niin templan puolelta pitää poistaa samalla `[formControl]="control"`.

Kuten juteltu, niin nää validointihommat tulee elämään, niin voi olla että näitä jossain vaiheessa tarvitsee tai sitten ei, mutta itse suosisin, että poistetaan kaikki jota ei tällä hetkellä tarvita, ettei ne vahingossa jää roikkumaan :bless:

Reply:
Joo siis mä jätin nää just siks et tää ois ns. "valmis" kun se validointi on tehty.